fcrepo-kernel-impl/src/test/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImplTest.java

```diff
@@ -0,0 +1,264 @@
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl.services;
+
+import static org.fcrepo.kernel.api.RdfLexicon.BASIC_CONTAINER;
+import static org.fcrepo.kernel.api.RdfLexicon.NON_RDF_SOURCE;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.util.ReflectionTestUtils.setField;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.fcrepo.kernel.api.exception.CannotCreateResourceException;
+import org.fcrepo.kernel.api.exception.ItemNotFoundException;
+import org.fcrepo.kernel.api.models.ExternalContent;
+import org.fcrepo.kernel.api.models.ResourceHeaders;
+import org.fcrepo.kernel.api.operations.NonRdfSourceOperationFactory;
+import org.fcrepo.kernel.api.operations.RdfSourceOperation;
+import org.fcrepo.kernel.api.operations.RdfSourceOperationBuilder;
+import org.fcrepo.kernel.api.operations.RdfSourceOperationFactory;
+import org.fcrepo.kernel.api.operations.ResourceOperation;
+import org.fcrepo.kernel.api.services.functions.UniqueValueSupplier;
+import org.fcrepo.kernel.impl.operations.NonRdfSourceOperationFactoryImpl;
+import org.fcrepo.kernel.impl.operations.RdfSourceOperationFactoryImpl;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+import org.fcrepo.persistence.api.PersistentStorageSessionManager;
+import org.fcrepo.persistence.api.exceptions.PersistentItemNotFoundException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class CreateResourceServiceImplTest {
+
+    @Mock
+    private PersistentStorageSessionManager psManager;
+
+    private RdfSourceOperationFactory rdfSourceOperationFactory;
+
+    private NonRdfSourceOperationFactory nonRdfSourceOperationFactory;
+
+    @Mock
+    private UniqueValueSupplier minter;
+
+    @Mock
+    private PersistentStorageSession psSession;
+
+    @Mock
+    private ResourceHeaders resourceHeaders;
+
+    @Mock
+    private RdfSourceOperationBuilder builder;
+
+    @Mock
+    private RdfSourceOperation operation;
+
+    @Mock
+    private ExternalContent extContent;
+
+    @InjectMocks
+    private CreateResourceServiceImpl createResourceService;
+
+    private final static String txId = "tx1234";
+
+    private final static Collection<String> nonRdfSourceTypes;
+
+    private final static Collection<String> basicContainerTypes;
+
+    private final Model model = ModelFactory.createDefaultModel();
+
+    private final Collection<String> digests = Stream.of("urn:sha1:12345abced")
```

Comment:
Since the implementation doesn't matter here, I'd suggest replacing this with `Collections.singleton`